### ISSUE TITLE
Syntax highlighting for nested expression statements

### ISSUE THREAD

**Issue by jsiebern** [jaredly/reason-language-server#80]
Hi, I'm not sure if this is even part of the language server so please close this if it's not :)

When ever I use an expression like `[%mui.withStyles ...]` nested inside a module, the syntax highlighting fails. It works if it is on the base level, see this example:

<img width="728" alt="dashboard_mailsettings_re_ _frontend" src="https://user-images.githubusercontent.com/6476589/44086764-dfe96b60-9fbd-11e8-816e-ce85653dc21b.png">

If this doesn't come from the lsp, could you point me in the direction of where I'd need to report this? Because the highlighting is completely gone when the lsp isn't installed, that's why I hope it's relevant!

**Comment by jaredly** [comment]
Yeah, the .vsix vsode plugin contains the syntax highlighting logic [here](https://github.com/jaredly/reason-language-server/blob/master/client/reason.json). I copied it straight over from https://github.com/reasonml-editor/vscode-reasonml, so I don't know a ton about it :P
but yes, this is the correct place to report it

**Comment by texastoland** [comment]
I was thinking of posting a higher level issue. Have you seen @glennsl's [sane-reason-grammar](https://github.com/glennsl/sane-reason-grammar) fork? Would it make sense to list in `extensionDependencies` and maintain it independently?

**Comment by jaredly** [comment]
ooh, that does look quite nice. I think it probably makes most sense to bring everything into a monorepo, tbh. (all the editor plugins, and the language server, and the syntax definitions)

**Comment by texastoland** [comment]
👍🏽 I've written grammars too if someone needs resources later. My #1 advice is develop in YAML and build to JSON using the [Code extension](https://marketplace.visualstudio.com/items?itemName=Togusa09.tmlanguage) or [ST3 Package Dev](https://packagecontrol.io/packages/PackageDev).
